config/variables.cfg

```diff
@@ -47,19 +50,21 @@ GO_LATEST_TESTED="1.20.7"
-                RELEASENOTES=$(curl --silent  "https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/releases/latest" | grep '"body": '| cut -f2- -d:)
-                LATEST_SCRIPTS_COMMIT=$(curl --silent "https://api.github.com/repos/multiversx/mx-chain-scripts/commits/master" | jq -r .sha)
-                GITHUB_RATE_LIMIT_RESPONSE=$(curl -s https://api.github.com/rate_limit)
-                TEMP_ENCODED_GO_VERSION=$(curl --silent "https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/contents/goVersion?ref=$CONFIGVER" | jq -r '.content')
-        else
-          CONFIGVER="tags/$(curl --silent -H "Authorization: token $GITHUBTOKEN" "https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')"
-          RELEASENOTES=$(curl --silent -H "Authorization: token $GITHUBTOKEN" "https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/releases/latest" | grep '"body": '| cut -f2- -d:)
-          LATEST_SCRIPTS_COMMIT=$(curl --silent -H "Authorization: token $GITHUBTOKEN" "https://api.github.com/repos/multiversx/mx-chain-scripts/commits/master" | jq -r .sha)
-          GITHUB_RATE_LIMIT_RESPONSE=$(curl --silent -H "Authorization: token $GITHUBTOKEN" "https://api.github.com/rate_limit")
-          TEMP_ENCODED_GO_VERSION=$(curl --silent -H "Authorization: token $GITHUBTOKEN" "https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/contents/goVersion?ref=$CONFIGVER" | jq -r '.content')
-    fi
+# Define the authorization header if GITHUBTOKEN is available
+AUTH_HEADER=""
+[ -n "$GITHUBTOKEN" ] && AUTH_HEADER="-H \"Authorization: $GITHUBTOKEN\""
+
+# Fetch latest config version and release notes
+CONFIGVER="tags/$(eval curl --silent $AUTH_HEADER \"https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/releases/latest\" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')"
+RELEASENOTES=$(eval curl --silent $AUTH_HEADER \"https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/releases/latest\" | grep '"body": '| cut -f2- -d:)
+
+# Get the latest scripts commit and rate limit response
+LATEST_SCRIPTS_COMMIT=$(eval curl --silent $AUTH_HEADER \"https://api.github.com/repos/multiversx/mx-chain-scripts/commits/master\" | jq -r .sha)
+GITHUB_RATE_LIMIT_RESPONSE=$(eval curl --silent $AUTH_HEADER \"https://api.github.com/rate_limit\")
+
+# Determine the goVersion to fetch
+API_CALL_CONFIGVER=${OVERRIDE_CONFIGVER:-$CONFIGVER}
+TEMP_ENCODED_GO_VERSION=$(eval curl --silent $AUTH_HEADER \"https://api.github.com/repos/multiversx/mx-chain-$ENVIRONMENT-config/contents/goVersion?ref=$API_CALL_CONFIGVER\" | jq -r '.content')
 
 # TODO: remove this if once all chain configs have goVersion file and update TEMP_ENCODED_GO_VERSION
```

Comment:
this TODO can be solved/removed, as all chains now have goVersion file on the latest release